src/main/java/com/uid2/admin/vertx/service/SiteService.java

```diff
@@ -72,15 +72,36 @@ public void setupRoutes(Router router) {
             }
         }, Role.CLIENTKEY_ISSUER));
         router.post("/api/site/set-types").blockingHandler(auth.handle((ctx) -> {
-                    synchronized (writeLock) {
-                        this.handleSiteTypesSet(ctx);
-                    }
-                    }, Role.CLIENTKEY_ISSUER));
+            synchronized (writeLock) {
+                this.handleSiteTypesSet(ctx);
+            }
+        }, Role.CLIENTKEY_ISSUER));
         router.post("/api/site/domain_names").blockingHandler(auth.handle((ctx) -> {
             synchronized (writeLock) {
                 this.handleSiteDomains(ctx);
             }
         }, Role.CLIENTKEY_ISSUER));
+        handleSiteBackfillCreated();
+    }
+
+    private void handleSiteBackfillCreated() {
+        try {
+            final Collection<Site> sites = new ArrayList<>(this.siteProvider.getAllSites());
+            final Collection<Site> updatedSites = new ArrayList<>();
+            for (Site s : sites) {
+                ClientKey key = this.clientKeyProvider.getOldestClientKey(s.getId());
+                if (s.getCreated() == 0 && key != null) {
+                    long created = key.getCreated();
+                    Site us = new Site(s.getId(), s.getName(), s.isEnabled(), s.getClientTypes(), s.getDomainNames(), created);
+                    updatedSites.add(us);
+                } else {
```

Comment:
Do you think its worth it to log sites where the `created` timestamp is 0 as well as there are no valid client keys ? Or you have from production sites.json that this scenario doesn't exist ?

Reply:
It may be possible for sites to exist without a client key like if one created one in testing, but since we can't delete sites I don't think there is a point in listing them.